src/backend.c

```diff
@@ -72,7 +72,12 @@ accept_backend_arg(struct Backend *backend, const char *arg) {
         }
 #endif
     } else if (address_port(backend->address) == 0 && is_numeric(arg)) {
-        address_set_port(backend->address, atoi(arg));
+        int port = atoi(arg);
+        if (port < 0 || port > 65535) {
+            err("Invalid port: %s", arg);
+            return -1;
+        }
```

Comment:
Let's pull this validation logic into a helper function in the address module, maybe `address_set_port_str()`?